# CSV to Vdata

## Table of content

* 1-Import
* 2 - Ensembl id is unknown
* 3 - Proteins with at least two ensembl ids
* 4 - Several proteins for one gene
* 5 - Var dataframe
* 6- raw layer
* 7 - Mean layer
* 8 - replicates layers
* 9 - test VData reading

## 1- Import

In [1]:
import pandas as pd
import vdata
from vdata import TemporalDataFrame, VData, setLoggingLevel

Proteomic data from the step 1

In [3]:
proteomic_df = pd.read_csv(
    "/home/matteo/serveur_vidium/DMD/1-ExperimentalData/Proteomic_data/steps/01-Protein_to_ENSEMBL_ID/results"
    "/Proteomic_data_ENSEMBL_ID_curated.csv",
    sep=",",
    index_col="id",
)

In [4]:
proteomic_df.head()

All_protein_ID  \
id                                                      
0                                   Q9H0G5;A0A024QZ33   
1   P35237;A0A2R8YD12;A0A024QZX5;A0A087X1N8;A0A2R8...   
2                            Q00341;A0A024R4E5;H0Y394   
3                     Q9H4M9;A0A024R571;C9JC03;C9J2Z4   
4                        P25787;A0A024RA52;A0A7I2V2H3   

                                        Protein names    Gene names  \
id                                                                    
0       Nuclear speckle splicing regulatory protein 1  NSRP1;CCDC55   
1                                           Serpin B6      SERPINB6   
2                                             Vigilin         HDLBP   
3                      EH domain-containing protein 1          EHD1   
4   Proteasome subunit alpha type-2;Proteasome sub...         PSMA2   

   Mol. weight [kDa]  Razor + unique peptides R1  Razor + unique peptides R2  \
id                                                                             
0             66,389                           1                           2   
1             42,621                           2                           0   
2             141,45                          27                          23   
3             60,626                          17                          16   
4             25,898                           9                           8   

    Razor + unique peptides R3 Unique + razor sequence coverage [%]  \
id                                                                    
0                            2                                    5   
1                            0                                  9,3   
2                           28                                 35,1   
3                           13                                 51,7   
4                            7                                 35,9   

   ANOVA Significant (1% BH-FDR)) ANOVA q-value  ... rawR3-WT9 rawR3-Mut9  \
id                                               ...                        
0                             NaN      9,25E-01  ...     92848     112510   
1                             NaN           NaN  ...       NaN        NaN   
2                             NaN      1,00E+00  ...   1764200    1653100   
3                               +      1,01E-03  ...    679770     588170   
4                             NaN      1,00E+00  ...    459200     519630   

   rawR3-WT10 rawR3-Mut10 rawR3-WT12 rawR3-Mut12 rawR3-WT14 rawR3-Mut14  \
id                                                                        
0      136290      128080     118650      134610     214620      133270   
1         NaN         NaN        NaN         NaN        NaN         NaN   
2     1988400     1844300    1854200     1677400    2280200     2400300   
3      776850      637960     752200      645690     848520      978810   
4      484740      598120     485480      569490     638210      617400   

                     All_ENSEMBL_ID  Main_ensembl_id  
id                                                    
0                   ENSG00000126653  ENSG00000126653  
1                   ENSG00000124570  ENSG00000124570  
2                   ENSG00000115677  ENSG00000115677  
3                   ENSG00000110047  ENSG00000110047  
4   ENSG00000106588;ENSG00000256646  ENSG00000106588  

[5 rows x 124 columns]

We only keep raw count data.

In [5]:
proteomic_df = proteomic_df[
    proteomic_df.columns[
        ((proteomic_df.columns.str[0:3] == "raw") == True)
        | ((proteomic_df.columns.str[:] == "Main_ensembl_id") == True)
        | ((proteomic_df.columns.str[:] == "Gene names") == True)
    ]
]
proteomic_df = proteomic_df.apply(lambda x: x.str.replace(",", "."))
proteomic_df[
    proteomic_df.columns[(proteomic_df.columns.str[0:3] == "raw") == True]
] = proteomic_df[
    proteomic_df.columns[(proteomic_df.columns.str[0:3] == "raw") == True].tolist()
].astype(
    float
)

proteomic_df.head()

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  rawR1-WT7  \
id                                                                          
0   NSRP1;CCDC55   112150.0    100250.0   102800.0    115760.0    93172.0   
1       SERPINB6    40426.0     34249.0    44000.0     43406.0    33604.0   
2          HDLBP  1351100.0   1226800.0  1524000.0   1585600.0  1331200.0   
3           EHD1   554030.0    553380.0   627240.0    689730.0   628120.0   
4          PSMA2   632690.0    614100.0   690080.0    727490.0   587610.0   

    rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  rawR3-Mut8  rawR3-WT9  \
id                                                ...                          
0     115860.0   105610.0    106860.0    94802.0  ...     96349.0    92848.0   
1      38418.0    42710.0     39058.0    43591.0  ...         NaN        NaN   
2    1535400.0  1618700.0   1366300.0  1442700.0  ...   1697200.0  1764200.0   
3     708150.0   795010.0    671280.0   783310.0  ...    590670.0   679770.0   
4     731020.0   662280.0    687820.0   592640.0  ...    540440.0   459200.0   

    rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  rawR3-WT14  \
id                                                                             
0     112510.0    136290.0     128080.0    118650.0     134610.0    214620.0   
1          NaN         NaN          NaN         NaN          NaN         NaN   
2    1653100.0   1988400.0    1844300.0   1854200.0    1677400.0   2280200.0   
3     588170.0    776850.0     637960.0    752200.0     645690.0    848520.0   
4     519630.0    484740.0     598120.0    485480.0     569490.0    638210.0   

    rawR3-Mut14  Main_ensembl_id  
id                                
0      133270.0  ENSG00000126653  
1           NaN  ENSG00000124570  
2     2400300.0  ENSG00000115677  
3      978810.0  ENSG00000110047  
4      617400.0  ENSG00000106588  

[5 rows x 50 columns]

In [6]:
proteomic_df = proteomic_df.rename(
    columns={
        "rawR1-Mut14": "rawR1-WT14",
        "rawR2-Mut14": "rawR2-WT14",
        "rawR3-Mut14": "rawR3-WT14",
        "rawR1-WT14": "rawR1-Mut14",
        "rawR2-WT14": "rawR2-Mut14",
        "rawR3-WT14": "rawR3-Mut14",
    }
)

There are few ambiguous cases :
- One row (so one protein ) could have several gene names and ensembl ids. In this cases we think that we can't trust the protein identification, so we remove rows.
- Several proteins can share one gene in common. As our analysis are focus on gene level, we chose to add counts of proteins sharing sames genes.
- The ensembl id is unknown


### 2 - Ensembl id is unknown

For these 3 genes the main ensembl id (based on hgnc is unknown), we removes rows

In [7]:
proteomic_df[proteomic_df.Main_ensembl_id.isna() == True]

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  rawR1-WT7  \
id                                                                            
422   HLA-C;HLA-Cw    25236.0     34094.0    22327.0     37721.0    19715.0   
423          HLA-A    96311.0     48499.0    99515.0     56060.0    85684.0   
5548         L1RE1    11675.0      8410.5    11608.0     12965.0     8512.3   

      rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  rawR3-Mut8  \
id                                                  ...               
422      32602.0    28897.0     31499.0    23572.0  ...     24989.0   
423      54221.0    97329.0     55649.0    89772.0  ...     36288.0   
5548     11197.0    10573.0     11944.0    11624.0  ...     10395.0   

      rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  \
id                                                                              
422     20656.0     26159.0     24164.0      30933.0     24456.0      26834.0   
423     66538.0     35510.0     78042.0      45924.0     72245.0      43213.0   
5548    13198.0     11408.0     14920.0      12642.0     14824.0      14101.0   

      rawR3-Mut14  rawR3-WT14  Main_ensembl_id  
id                                              
422       42252.0     26472.0              NaN  
423       61664.0     84730.0              NaN  
5548      18759.0     17082.0              NaN  

[3 rows x 50 columns]

In [8]:
proteomic_df = proteomic_df[proteomic_df.Main_ensembl_id.isna() == False]
proteomic_df

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  \
id                                                                     
0         NSRP1;CCDC55   112150.0    100250.0   102800.0    115760.0   
1             SERPINB6    40426.0     34249.0    44000.0     43406.0   
2                HDLBP  1351100.0   1226800.0  1524000.0   1585600.0   
3                 EHD1   554030.0    553380.0   627240.0    689730.0   
4                PSMA2   632690.0    614100.0   690080.0    727490.0   
...                ...        ...         ...        ...         ...   
5738          IVNS1ABP    95734.0    117470.0   105900.0    138760.0   
5739           SEC23IP   690970.0    678470.0   754330.0    835280.0   
5740  COMMD3-BMI1;BMI1    74062.0     73028.0    77998.0     80546.0   
5819             NOLC1    22587.0     24340.0    24021.0     28617.0   
5820            USP6NL        NaN         NaN        NaN         NaN   

      rawR1-WT7  rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  \
id                                                             ...   
0       93172.0    115860.0   105610.0    106860.0    94802.0  ...   
1       33604.0     38418.0    42710.0     39058.0    43591.0  ...   
2     1331200.0   1535400.0  1618700.0   1366300.0  1442700.0  ...   
3      628120.0    708150.0   795010.0    671280.0   783310.0  ...   
4      587610.0    731020.0   662280.0    687820.0   592640.0  ...   
...         ...         ...        ...         ...        ...  ...   
5738    85028.0    106300.0   105030.0    101820.0    76005.0  ...   
5739   635770.0    770690.0   737600.0    692510.0   689790.0  ...   
5740    69430.0     86929.0    85239.0     79020.0    81838.0  ...   
5819    23234.0     28480.0    22604.0     26211.0    23737.0  ...   
5820        NaN         NaN        NaN         NaN        NaN  ...   

      rawR3-Mut8  rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  \
id                                                                             
0        96349.0    92848.0    112510.0    136290.0     128080.0    118650.0   
1            NaN        NaN         NaN         NaN          NaN         NaN   
2      1697200.0  1764200.0   1653100.0   1988400.0    1844300.0   1854200.0   
3       590670.0   679770.0    588170.0    776850.0     637960.0    752200.0   
4       540440.0   459200.0    519630.0    484740.0     598120.0    485480.0   
...          ...        ...         ...         ...          ...         ...   
5738    103620.0    79397.0    108470.0     91254.0     120470.0     81211.0   
5739    637280.0   583910.0    642950.0    749620.0     735570.0    692760.0   
5740    106750.0   102060.0    120740.0    154340.0     135800.0    136330.0   
5819     11943.0    11628.0     12508.0     15604.0      13009.0     13679.0   
5820     20682.0    20600.0     18146.0     20911.0      20950.0     17867.0   

      rawR3-Mut12  rawR3-Mut14  rawR3-WT14                  Main_ensembl_id  
id                                                                           
0        134610.0     214620.0    133270.0                  ENSG00000126653  
1             NaN          NaN         NaN                  ENSG00000124570  
2       1677400.0    2280200.0   2400300.0                  ENSG00000115677  
3        645690.0     848520.0    978810.0                  ENSG00000110047  
4        569490.0     638210.0    617400.0                  ENSG00000106588  
...           ...          ...         ...                              ...  
5738     115940.0     129670.0     93933.0                  ENSG00000116679  
5739     730010.0    1019000.0    851350.0                  ENSG00000107651  
5740     139650.0     209550.0    147950.0  ENSG00000269897;ENSG00000168283  
5819      13382.0      18198.0     13243.0                  ENSG00000166197  
5820      19945.0      23489.0     23225.0                  ENSG00000148429  

[5632 rows x 50 columns]

## 3 - Proteins with at least two ensembl ids
We think we cannot distinguish proteins with the proteomic experiment and aren't confident. So we remove these rows.

In [9]:
proteomic_gene_df = proteomic_df[["Main_ensembl_id"]].astype(str)
proteomic_gene_df

Main_ensembl_id
id                                   
0                     ENSG00000126653
1                     ENSG00000124570
2                     ENSG00000115677
3                     ENSG00000110047
4                     ENSG00000106588
...                               ...
5738                  ENSG00000116679
5739                  ENSG00000107651
5740  ENSG00000269897;ENSG00000168283
5819                  ENSG00000166197
5820                  ENSG00000148429

[5632 rows x 1 columns]

We want to check if genes with multiple ensembl ids have multiple gene names

In [10]:
def check_multi_gene(row, list_multi_gene_prot, name_col):
    list_gene = row[name_col].split(";")
    if len(list_gene) > 1:
        list_multi_gene_prot.append(row.id)
    return

In [11]:
list_multi_ensembl = []
proteomic_gene_df.reset_index().apply(
    lambda row: check_multi_gene(row, list_multi_ensembl, "Main_ensembl_id"), axis=1
)
len(list_multi_ensembl)

218

There are 218 proteins with at least two ensembl id

In [12]:
prot_df = proteomic_df.reset_index()[["id", "Gene names"]]
prot_df = prot_df.dropna()
list_multi_gene_name = []
prot_df.apply(
    lambda row: check_multi_gene(row, list_multi_gene_name, "Gene names"), axis=1
)
len(list_multi_gene_name)

264

There are 264 protein with at least 2 gene names

List of proteins with multiple ensembl ids but only one gene name.

In [13]:
set(list_multi_ensembl) - set(list_multi_gene_name)

{193, 412, 722, 806, 1061, 2044, 2675, 2676, 2881, 3393, 4182, 5515, 5630}

We remove genes with several ensembl ids

In [14]:
proteomic_df = proteomic_df[proteomic_df.index.isin(list_multi_ensembl) == False]
proteomic_df

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  rawR1-WT7  \
id                                                                            
0     NSRP1;CCDC55   112150.0    100250.0   102800.0    115760.0    93172.0   
1         SERPINB6    40426.0     34249.0    44000.0     43406.0    33604.0   
2            HDLBP  1351100.0   1226800.0  1524000.0   1585600.0  1331200.0   
3             EHD1   554030.0    553380.0   627240.0    689730.0   628120.0   
4            PSMA2   632690.0    614100.0   690080.0    727490.0   587610.0   
...            ...        ...         ...        ...         ...        ...   
5736       FAM169A    86450.0     81010.0    79561.0     87633.0    72422.0   
5738      IVNS1ABP    95734.0    117470.0   105900.0    138760.0    85028.0   
5739       SEC23IP   690970.0    678470.0   754330.0    835280.0   635770.0   
5819         NOLC1    22587.0     24340.0    24021.0     28617.0    23234.0   
5820        USP6NL        NaN         NaN        NaN         NaN        NaN   

      rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  rawR3-Mut8  \
id                                                  ...               
0       115860.0   105610.0    106860.0    94802.0  ...     96349.0   
1        38418.0    42710.0     39058.0    43591.0  ...         NaN   
2      1535400.0  1618700.0   1366300.0  1442700.0  ...   1697200.0   
3       708150.0   795010.0    671280.0   783310.0  ...    590670.0   
4       731020.0   662280.0    687820.0   592640.0  ...    540440.0   
...          ...        ...         ...        ...  ...         ...   
5736     80007.0    86796.0     66521.0    67492.0  ...    161910.0   
5738    106300.0   105030.0    101820.0    76005.0  ...    103620.0   
5739    770690.0   737600.0    692510.0   689790.0  ...    637280.0   
5819     28480.0    22604.0     26211.0    23737.0  ...     11943.0   
5820         NaN        NaN         NaN        NaN  ...     20682.0   

      rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  \
id                                                                              
0       92848.0    112510.0    136290.0     128080.0    118650.0     134610.0   
1           NaN         NaN         NaN          NaN         NaN          NaN   
2     1764200.0   1653100.0   1988400.0    1844300.0   1854200.0    1677400.0   
3      679770.0    588170.0    776850.0     637960.0    752200.0     645690.0   
4      459200.0    519630.0    484740.0     598120.0    485480.0     569490.0   
...         ...         ...         ...          ...         ...          ...   
5736   158640.0    148710.0    185620.0     156070.0    146580.0     155410.0   
5738    79397.0    108470.0     91254.0     120470.0     81211.0     115940.0   
5739   583910.0    642950.0    749620.0     735570.0    692760.0     730010.0   
5819    11628.0     12508.0     15604.0      13009.0     13679.0      13382.0   
5820    20600.0     18146.0     20911.0      20950.0     17867.0      19945.0   

      rawR3-Mut14  rawR3-WT14  Main_ensembl_id  
id                                              
0        214620.0    133270.0  ENSG00000126653  
1             NaN         NaN  ENSG00000124570  
2       2280200.0   2400300.0  ENSG00000115677  
3        848520.0    978810.0  ENSG00000110047  
4        638210.0    617400.0  ENSG00000106588  
...           ...         ...              ...  
5736     206360.0    179540.0  ENSG00000198780  
5738     129670.0     93933.0  ENSG00000116679  
5739    1019000.0    851350.0  ENSG00000107651  
5819      18198.0     13243.0  ENSG00000166197  
5820      23489.0     23225.0  ENSG00000148429  

[5414 rows x 50 columns]

Now we have 5414 proteins

### 4 - Several proteins for one gene

In [15]:
list_dupli_ens_id = (
    proteomic_df[proteomic_df.Main_ensembl_id.duplicated(keep=False) == True]
    .Main_ensembl_id.unique()
    .tolist()
)
len(list_dupli_ens_id)

52

There are 52 duplicated genes. One example below.

In [16]:
proteomic_df[proteomic_df.Main_ensembl_id == "ENSG00000143549"]

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  rawR1-WT7  \
id                                                                         
88         TPM3  1839100.0   1692700.0  1592800.0   1714300.0  1438900.0   
1944       TPM3    74029.0     71527.0    62834.0     64963.0    60511.0   

      rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  rawR3-Mut8  \
id                                                  ...               
88     1714100.0  1690100.0   1568600.0  1525500.0  ...   1941400.0   
1944     77468.0    53482.0     60553.0    52209.0  ...     54626.0   

      rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  \
id                                                                              
88    1875200.0   1927900.0   2773400.0    2092200.0   2108800.0    2017400.0   
1944    48172.0     61561.0     66844.0      74669.0     51672.0      67003.0   

      rawR3-Mut14  rawR3-WT14  Main_ensembl_id  
id                                              
88      3308900.0   2741800.0  ENSG00000143549  
1944      76524.0     65664.0  ENSG00000143549  

[2 rows x 50 columns]

We separate unique and duplicated genes.

In [17]:
proteomic_df_unique = proteomic_df[
    proteomic_df.Main_ensembl_id.isin(list_dupli_ens_id) == False
]
proteomic_df_unique

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  rawR1-WT7  \
id                                                                            
0     NSRP1;CCDC55   112150.0    100250.0   102800.0    115760.0    93172.0   
1         SERPINB6    40426.0     34249.0    44000.0     43406.0    33604.0   
2            HDLBP  1351100.0   1226800.0  1524000.0   1585600.0  1331200.0   
3             EHD1   554030.0    553380.0   627240.0    689730.0   628120.0   
4            PSMA2   632690.0    614100.0   690080.0    727490.0   587610.0   
...            ...        ...         ...        ...         ...        ...   
5735         WASF2   167980.0    170560.0   176340.0    191240.0   159770.0   
5736       FAM169A    86450.0     81010.0    79561.0     87633.0    72422.0   
5738      IVNS1ABP    95734.0    117470.0   105900.0    138760.0    85028.0   
5739       SEC23IP   690970.0    678470.0   754330.0    835280.0   635770.0   
5820        USP6NL        NaN         NaN        NaN         NaN        NaN   

      rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  rawR3-Mut8  \
id                                                  ...               
0       115860.0   105610.0    106860.0    94802.0  ...     96349.0   
1        38418.0    42710.0     39058.0    43591.0  ...         NaN   
2      1535400.0  1618700.0   1366300.0  1442700.0  ...   1697200.0   
3       708150.0   795010.0    671280.0   783310.0  ...    590670.0   
4       731020.0   662280.0    687820.0   592640.0  ...    540440.0   
...          ...        ...         ...        ...  ...         ...   
5735    191140.0   201580.0    177900.0   164940.0  ...    285430.0   
5736     80007.0    86796.0     66521.0    67492.0  ...    161910.0   
5738    106300.0   105030.0    101820.0    76005.0  ...    103620.0   
5739    770690.0   737600.0    692510.0   689790.0  ...    637280.0   
5820         NaN        NaN         NaN        NaN  ...     20682.0   

      rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  \
id                                                                              
0       92848.0    112510.0    136290.0     128080.0    118650.0     134610.0   
1           NaN         NaN         NaN          NaN         NaN          NaN   
2     1764200.0   1653100.0   1988400.0    1844300.0   1854200.0    1677400.0   
3      679770.0    588170.0    776850.0     637960.0    752200.0     645690.0   
4      459200.0    519630.0    484740.0     598120.0    485480.0     569490.0   
...         ...         ...         ...          ...         ...          ...   
5735   251390.0    271040.0    357900.0     317720.0    282600.0     296670.0   
5736   158640.0    148710.0    185620.0     156070.0    146580.0     155410.0   
5738    79397.0    108470.0     91254.0     120470.0     81211.0     115940.0   
5739   583910.0    642950.0    749620.0     735570.0    692760.0     730010.0   
5820    20600.0     18146.0     20911.0      20950.0     17867.0      19945.0   

      rawR3-Mut14  rawR3-WT14  Main_ensembl_id  
id                                              
0        214620.0    133270.0  ENSG00000126653  
1             NaN         NaN  ENSG00000124570  
2       2280200.0   2400300.0  ENSG00000115677  
3        848520.0    978810.0  ENSG00000110047  
4        638210.0    617400.0  ENSG00000106588  
...           ...         ...              ...  
5735     433370.0    341080.0  ENSG00000158195  
5736     206360.0    179540.0  ENSG00000198780  
5738     129670.0     93933.0  ENSG00000116679  
5739    1019000.0    851350.0  ENSG00000107651  
5820      23489.0     23225.0  ENSG00000148429  

[5308 rows x 50 columns]

In [18]:
for ens_id in list_dupli_ens_id:  # for each gene in the list of duplicated gene
    dupli_id_df = proteomic_df[(proteomic_df.Main_ensembl_id == ens_id) == True]
    # We sum protein counts
    dupli_id_df_sum = pd.DataFrame(
        dupli_id_df[
            dupli_id_df.columns[((dupli_id_df.columns.str[0:3] == "raw") == True)]
        ].sum()
    ).transpose()

    dupli_id_df_concat = pd.concat(
        [
            dupli_id_df[["Gene names", "Main_ensembl_id"]][:1].reset_index(),
            dupli_id_df_sum,
        ],
        axis=1,
        join="outer",
    )
    id_multi = ""
    for k in dupli_id_df.index.tolist():
        id_multi = id_multi + str(k) + ";"
    id_multi[:-1]
    dupli_id_df_concat.loc[:, "id"] = id_multi[:-1]
    dupli_id_df_concat = dupli_id_df_concat.set_index("id")
    proteomic_df_unique = pd.concat([proteomic_df_unique, dupli_id_df_concat], axis=0)

In [19]:
dupli_id_df

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  rawR1-WT7  \
id                                                                         
3494       CAP1  2015300.0   1742100.0  2092900.0   2151900.0  1894300.0   
4000       CAP1    32545.0     35966.0    39713.0     42326.0    33062.0   

      rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  rawR3-Mut8  \
id                                                  ...               
3494   2207200.0  2109600.0   1990700.0  1832500.0  ...   2168700.0   
4000     36922.0    36225.0     39256.0    34726.0  ...     29139.0   

      rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  \
id                                                                              
3494  2002000.0   2009400.0   2156700.0    2236900.0   2014800.0    2137300.0   
4000    25864.0     29423.0     34419.0      31999.0     30346.0      30583.0   

      rawR3-Mut14  rawR3-WT14  Main_ensembl_id  
id                                              
3494    2639800.0   2653200.0  ENSG00000131236  
4000      44862.0     41314.0  ENSG00000131236  

[2 rows x 50 columns]

In [20]:
proteomic_df_unique

Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  rawR1-Mut6  \
id                                                                      
0          NSRP1;CCDC55   112150.0    100250.0   102800.0    115760.0   
1              SERPINB6    40426.0     34249.0    44000.0     43406.0   
2                 HDLBP  1351100.0   1226800.0  1524000.0   1585600.0   
3                  EHD1   554030.0    553380.0   627240.0    689730.0   
4                 PSMA2   632690.0    614100.0   690080.0    727490.0   
...                 ...        ...         ...        ...         ...   
2630;4028          TPM2  2149079.8   1599522.0  2067257.2   1593945.1   
3074;3075          TMPO  1366240.0   1240150.0  1310350.0   1359200.0   
3281;4053         PSMD4   331080.0    296615.0   322474.0    349703.0   
3362;4016        HNRNPK  4932500.0   4724920.0  4736420.0   5148430.0   
3494;4000          CAP1  2047845.0   1778066.0  2132613.0   2194226.0   

           rawR1-WT7  rawR1-Mut7  rawR1-WT8  rawR1-Mut8  rawR1-WT9  ...  \
id                                                                  ...   
0            93172.0    115860.0   105610.0    106860.0    94802.0  ...   
1            33604.0     38418.0    42710.0     39058.0    43591.0  ...   
2          1331200.0   1535400.0  1618700.0   1366300.0  1442700.0  ...   
3           628120.0    708150.0   795010.0    671280.0   783310.0  ...   
4           587610.0    731020.0   662280.0    687820.0   592640.0  ...   
...              ...         ...        ...         ...        ...  ...   
2630;4028  1968964.5   1954417.0  2193545.8   1818441.5  2032629.7  ...   
3074;3075  1119360.0   1321010.0  1269360.0   1297450.0  1244880.0  ...   
3281;4053   280905.0    336718.0   324638.0    316498.0   296653.0  ...   
3362;4016  4114530.0   4933370.0  4656730.0   4689070.0  4110580.0  ...   
3494;4000  1927362.0   2244122.0  2145825.0   2029956.0  1867226.0  ...   

           rawR3-Mut8  rawR3-WT9  rawR3-Mut9  rawR3-WT10  rawR3-Mut10  \
id                                                                      
0             96349.0    92848.0    112510.0    136290.0     128080.0   
1                 NaN        NaN         NaN         NaN          NaN   
2           1697200.0  1764200.0   1653100.0   1988400.0    1844300.0   
3            590670.0   679770.0    588170.0    776850.0     637960.0   
4            540440.0   459200.0    519630.0    484740.0     598120.0   
...               ...        ...         ...         ...          ...   
2630;4028   1825918.2  2048413.5   1877639.7   3476546.6    2044167.5   
3074;3075   1463830.0  1382880.0   1584690.0   1887750.0    1843980.0   
3281;4053    458710.0   435810.0    447960.0    532100.0     513750.0   
3362;4016   5232630.0  4660410.0   5595000.0   6201300.0    6694990.0   
3494;4000   2197839.0  2027864.0   2038823.0   2191119.0    2268899.0   

           rawR3-WT12  rawR3-Mut12  rawR3-Mut14  rawR3-WT14  Main_ensembl_id  
id                                                                            
0            118650.0     134610.0     214620.0    133270.0  ENSG00000126653  
1                 NaN          NaN          NaN         NaN  ENSG00000124570  
2           1854200.0    1677400.0    2280200.0   2400300.0  ENSG00000115677  
3            752200.0     645690.0     848520.0    978810.0  ENSG00000110047  
4            485480.0     569490.0     638210.0    617400.0  ENSG00000106588  
...               ...          ...          ...         ...              ...  
2630;4028   2347192.2    1963483.3    3642082.0   3474800.0  ENSG00000198467  
3074;3075   1750390.0    1886770.0    2885060.0   1889890.0  ENSG00000120802  
3281;4053    469430.0     512010.0     745610.0    574590.0  ENSG00000159352  
3362;4016   5754640.0    6839780.0   10083560.0   6647100.0  ENSG00000165119  
3494;4000   2045146.0    2167883.0    2684662.0   2694514.0  ENSG00000131236  

[5360 rows x 50 columns]

We have 5360 proteins (5038 + 52)

## 5 - Var dataframe

In [21]:
proteomic_var_df = (
    proteomic_df_unique[["Main_ensembl_id", "Gene names"]]
    .astype(str)
    .reset_index()
    .set_index("Main_ensembl_id")
    .rename(columns={"Gene names": "short_name"})
)
proteomic_var_df.loc[:, "organism"] = "Homo sapiens"
proteomic_var_df

id    short_name      organism
Main_ensembl_id                                       
ENSG00000126653          0  NSRP1;CCDC55  Homo sapiens
ENSG00000124570          1      SERPINB6  Homo sapiens
ENSG00000115677          2         HDLBP  Homo sapiens
ENSG00000110047          3          EHD1  Homo sapiens
ENSG00000106588          4         PSMA2  Homo sapiens
...                    ...           ...           ...
ENSG00000198467  2630;4028          TPM2  Homo sapiens
ENSG00000120802  3074;3075          TMPO  Homo sapiens
ENSG00000159352  3281;4053         PSMD4  Homo sapiens
ENSG00000165119  3362;4016        HNRNPK  Homo sapiens
ENSG00000131236  3494;4000          CAP1  Homo sapiens

[5360 rows x 3 columns]

## 6 - raw layer

In [22]:
proteomic_raw_df = proteomic_df_unique.reset_index()
proteomic_raw_df = proteomic_raw_df.set_index("Main_ensembl_id")
proteomic_raw_df


id    Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  \
Main_ensembl_id                                                              
ENSG00000126653          0  NSRP1;CCDC55   112150.0    100250.0   102800.0   
ENSG00000124570          1      SERPINB6    40426.0     34249.0    44000.0   
ENSG00000115677          2         HDLBP  1351100.0   1226800.0  1524000.0   
ENSG00000110047          3          EHD1   554030.0    553380.0   627240.0   
ENSG00000106588          4         PSMA2   632690.0    614100.0   690080.0   
...                    ...           ...        ...         ...        ...   
ENSG00000198467  2630;4028          TPM2  2149079.8   1599522.0  2067257.2   
ENSG00000120802  3074;3075          TMPO  1366240.0   1240150.0  1310350.0   
ENSG00000159352  3281;4053         PSMD4   331080.0    296615.0   322474.0   
ENSG00000165119  3362;4016        HNRNPK  4932500.0   4724920.0  4736420.0   
ENSG00000131236  3494;4000          CAP1  2047845.0   1778066.0  2132613.0   

                 rawR1-Mut6  rawR1-WT7  rawR1-Mut7  rawR1-WT8  rawR1-Mut8  \
Main_ensembl_id                                                             
ENSG00000126653    115760.0    93172.0    115860.0   105610.0    106860.0   
ENSG00000124570     43406.0    33604.0     38418.0    42710.0     39058.0   
ENSG00000115677   1585600.0  1331200.0   1535400.0  1618700.0   1366300.0   
ENSG00000110047    689730.0   628120.0    708150.0   795010.0    671280.0   
ENSG00000106588    727490.0   587610.0    731020.0   662280.0    687820.0   
...                     ...        ...         ...        ...         ...   
ENSG00000198467   1593945.1  1968964.5   1954417.0  2193545.8   1818441.5   
ENSG00000120802   1359200.0  1119360.0   1321010.0  1269360.0   1297450.0   
ENSG00000159352    349703.0   280905.0    336718.0   324638.0    316498.0   
ENSG00000165119   5148430.0  4114530.0   4933370.0  4656730.0   4689070.0   
ENSG00000131236   2194226.0  1927362.0   2244122.0  2145825.0   2029956.0   

                 ...  rawR3-WT8  rawR3-Mut8  rawR3-WT9  rawR3-Mut9  \
Main_ensembl_id  ...                                                 
ENSG00000126653  ...    99239.0     96349.0    92848.0    112510.0   
ENSG00000124570  ...        NaN         NaN        NaN         NaN   
ENSG00000115677  ...  1984500.0   1697200.0  1764200.0   1653100.0   
ENSG00000110047  ...   692700.0    590670.0   679770.0    588170.0   
ENSG00000106588  ...   521490.0    540440.0   459200.0    519630.0   
...              ...        ...         ...        ...         ...   
ENSG00000198467  ...  2259976.7   1825918.2  2048413.5   1877639.7   
ENSG00000120802  ...  1465710.0   1463830.0  1382880.0   1584690.0   
ENSG00000159352  ...   486690.0    458710.0   435810.0    447960.0   
ENSG00000165119  ...  5144430.0   5232630.0  4660410.0   5595000.0   
ENSG00000131236  ...  2338682.0   2197839.0  2027864.0   2038823.0   

                 rawR3-WT10  rawR3-Mut10  rawR3-WT12  rawR3-Mut12  \
Main_ensembl_id                                                     
ENSG00000126653    136290.0     128080.0    118650.0     134610.0   
ENSG00000124570         NaN          NaN         NaN          NaN   
ENSG00000115677   1988400.0    1844300.0   1854200.0    1677400.0   
ENSG00000110047    776850.0     637960.0    752200.0     645690.0   
ENSG00000106588    484740.0     598120.0    485480.0     569490.0   
...                     ...          ...         ...          ...   
ENSG00000198467   3476546.6    2044167.5   2347192.2    1963483.3   
ENSG00000120802   1887750.0    1843980.0   1750390.0    1886770.0   
ENSG00000159352    532100.0     513750.0    469430.0     512010.0   
ENSG00000165119   6201300.0    6694990.0   5754640.0    6839780.0   
ENSG00000131236   2191119.0    2268899.0   2045146.0    2167883.0   

                 rawR3-Mut14  rawR3-WT14  
Main_ensembl_id                           
ENSG00000126653     214620.0    133270.0  
ENSG00000124570          NaN         NaN  
ENSG00000115677    2280200

## 7 - Mean layer

For WT cells

In [23]:
for cond in ["5", "6", "7", "8", "9", "10", "12", "14"]:
    proteomic_rep = proteomic_raw_df[
        proteomic_raw_df.columns[((proteomic_raw_df.columns.str[8:] == cond) == True)]
    ]

    # If just 1 value
    proteomic_rep_na = proteomic_rep[
        (proteomic_rep.isna().sum(axis=1) >= 2) == True
    ].copy()
    proteomic_rep_na.loc[:, "mean_WT" + cond] = "NaN"
    proteomic_rep_na = proteomic_rep_na[["mean_WT" + cond]]
    ## If 2 or 3 values
    proteomic_rep_val = proteomic_rep[
        (proteomic_rep.isna().sum(axis=1) <= 1) == True
    ].copy()
    proteomic_rep_val.loc[:, "mean_WT" + cond] = proteomic_rep_val.mean(axis=1)
    proteomic_rep_val = proteomic_rep_val[["mean_WT" + cond]]
    proteomic_rep_raw = pd.concat([proteomic_rep_val, proteomic_rep_na]).reset_index()
    proteomic_raw_df = proteomic_raw_df.merge(
        proteomic_rep_raw, left_on="Main_ensembl_id", right_on="Main_ensembl_id"
    ).set_index("Main_ensembl_id")

For Mutant cells

In [24]:
for cond in ["5", "6", "7", "8", "9", "10", "12", "14"]:
    proteomic_rep = proteomic_raw_df[
        proteomic_raw_df.columns[((proteomic_raw_df.columns.str[9:] == cond) == True)]
    ]

    # If just 1 value
    proteomic_rep_na = proteomic_rep[
        (proteomic_rep.isna().sum(axis=1) >= 2) == True
    ].copy()
    proteomic_rep_na.loc[:, "mean_Mu" + cond] = "NaN"
    proteomic_rep_na = proteomic_rep_na[["mean_Mu" + cond]]
    ## If 2 or 3 values
    proteomic_rep_val = proteomic_rep[
        (proteomic_rep.isna().sum(axis=1) <= 1) == True
    ].copy()
    proteomic_rep_val.loc[:, "mean_Mu" + cond] = proteomic_rep_val.mean(axis=1)
    proteomic_rep_val = proteomic_rep_val[["mean_Mu" + cond]]
    proteomic_rep_raw = pd.concat([proteomic_rep_val, proteomic_rep_na]).reset_index()
    proteomic_raw_df = proteomic_raw_df.merge(
        proteomic_rep_raw, left_on="Main_ensembl_id", right_on="Main_ensembl_id"
    ).set_index("Main_ensembl_id")

In [25]:
proteomic_raw_df

id    Gene names  rawR1-WT5  rawR1-Mut5  rawR1-WT6  \
Main_ensembl_id                                                              
ENSG00000126653          0  NSRP1;CCDC55   112150.0    100250.0   102800.0   
ENSG00000124570          1      SERPINB6    40426.0     34249.0    44000.0   
ENSG00000115677          2         HDLBP  1351100.0   1226800.0  1524000.0   
ENSG00000110047          3          EHD1   554030.0    553380.0   627240.0   
ENSG00000106588          4         PSMA2   632690.0    614100.0   690080.0   
...                    ...           ...        ...         ...        ...   
ENSG00000198467  2630;4028          TPM2  2149079.8   1599522.0  2067257.2   
ENSG00000120802  3074;3075          TMPO  1366240.0   1240150.0  1310350.0   
ENSG00000159352  3281;4053         PSMD4   331080.0    296615.0   322474.0   
ENSG00000165119  3362;4016        HNRNPK  4932500.0   4724920.0  4736420.0   
ENSG00000131236  3494;4000          CAP1  2047845.0   1778066.0  2132613.0   

                 rawR1-Mut6  rawR1-WT7  rawR1-Mut7  rawR1-WT8  rawR1-Mut8  \
Main_ensembl_id                                                             
ENSG00000126653    115760.0    93172.0    115860.0   105610.0    106860.0   
ENSG00000124570     43406.0    33604.0     38418.0    42710.0     39058.0   
ENSG00000115677   1585600.0  1331200.0   1535400.0  1618700.0   1366300.0   
ENSG00000110047    689730.0   628120.0    708150.0   795010.0    671280.0   
ENSG00000106588    727490.0   587610.0    731020.0   662280.0    687820.0   
...                     ...        ...         ...        ...         ...   
ENSG00000198467   1593945.1  1968964.5   1954417.0  2193545.8   1818441.5   
ENSG00000120802   1359200.0  1119360.0   1321010.0  1269360.0   1297450.0   
ENSG00000159352    349703.0   280905.0    336718.0   324638.0    316498.0   
ENSG00000165119   5148430.0  4114530.0   4933370.0  4656730.0   4689070.0   
ENSG00000131236   2194226.0  1927362.0   2244122.0  2145825.0   2029956.0   

                 ...       mean_WT12       mean_WT14        mean_Mu5  \
Main_ensembl_id  ...                                                   
ENSG00000126653  ...        125880.0   143576.666667    99497.666667   
ENSG00000124570  ...             NaN             NaN             NaN   
ENSG00000115677  ...       1528600.0  1929033.333333  1216723.333333   
ENSG00000110047  ...   811043.333333  1040583.333333   489693.333333   
ENSG00000106588  ...   522066.666667        663040.0   521756.666667   
...              ...             ...             ...             ...   
ENSG00000198467  ...       2028406.4  2988677.933333       1410389.9   
ENSG00000120802  ...       1671230.0  1784236.666667  1340403.333333   
ENSG00000159352  ...   378179.333333   461521.333333   340171.666667   
ENSG00000165119  ...  5211076.666667  6037593.333333  4854566.666667   
ENSG00000131236  ...  1924111.333333  2528436.333333  1840094.333333   

                       mean_Mu6        mean_Mu7        mean_Mu8  \
Main_ensembl_id                                                   
ENSG00000126653   121156.666667        115850.0   107396.333333   
ENSG00000124570             NaN             NaN             NaN   
ENSG00000115677       1568000.0  1511333.333333  1364966.666667   
ENSG00000110047        629270.0   658883.333333   610393.333333   
ENSG00000106588   620983.333333   626043.333333   585156.666667   
...                         ...             ...             ...   
ENSG00000198467       1424793.3  1739281.033333  1614441.966667   
ENSG00000120802  1458496.666667       1404260.0       1374850.0   
ENSG00000159352   396224.333333   380219.333333   365522.666667   
ENSG00000165119       5135550.0       4984850.0  4778246.666667   
ENSG00000131236  2226219.333333  2289256.333333       2051580.0   

                       mean_Mu9       mean_Mu10       mean_Mu12  \
Main_ensembl_id                                                   
ENSG00000126653        120460.0        137650.0   140793.333333 

In [26]:
raw_Mean_df = proteomic_raw_df[
    proteomic_raw_df.columns[(proteomic_raw_df.columns.str[:4] == "mean") == True]
]
raw_Mean_df = raw_Mean_df.transpose()
raw_Mean_df.index = [
    cond[5:7] for cond in list(raw_Mean_df.index)
]
raw_Mean_df.index = raw_Mean_df.index.rename("Cell_type")
raw_Mean_df


Main_ensembl_id ENSG00000126653 ENSG00000124570 ENSG00000115677  \
Cell_type                                                         
WT                115223.333333             NaN  1342333.333333   
WT                115546.666667             NaN       1521900.0   
WT                 98533.333333             NaN       1323300.0   
WT                     113793.0             NaN  1606833.333333   
WT                     104280.0             NaN  1435833.333333   
WT                148806.666667             NaN       1619400.0   
WT                     125880.0             NaN       1528600.0   
WT                143576.666667             NaN  1929033.333333   
Mu                 99497.666667             NaN  1216723.333333   
Mu                121156.666667             NaN       1568000.0   
Mu                     115850.0             NaN  1511333.333333   
Mu                107396.333333             NaN  1364966.666667   
Mu                     120460.0             NaN  1329723.333333   
Mu                     137650.0             NaN       1484700.0   
Mu                140793.333333             NaN  1361033.333333   
Mu                     238420.0             NaN  1850533.333333   

Main_ensembl_id ENSG00000110047 ENSG00000106588 ENSG00000011007  \
Cell_type                                                         
WT                519633.333333        529030.0    56806.666667   
WT                578723.333333   586206.666667    57049.333333   
WT                586043.333333   505453.333333    51996.333333   
WT                     725820.0   564583.333333         59662.0   
WT                717343.333333   502106.666667    53197.666667   
WT                820133.333333        526530.0    73354.666667   
WT                811043.333333   522066.666667    63800.333333   
WT               1040583.333333        663040.0    71504.666667   
Mu                489693.333333   521756.666667    53170.333333   
Mu                     629270.0   620983.333333    58397.666667   
Mu                658883.333333   626043.333333    60331.666667   
Mu                610393.333333   585156.666667    56645.666667   
Mu                608383.333333        563710.0         63414.0   
Mu                673286.666667   657853.333333         71928.0   
Mu                     693680.0   622463.333333    67810.666667   
Mu                     927080.0        696580.0   103134.333333   

Main_ensembl_id ENSG00000184047 ENSG00000204463 ENSG00000101974  \
Cell_type                                                         
WT                     128469.0   653866.666667         44157.5   
WT                147017.333333        704560.0         55676.5   
WT                119352.666667   572913.333333         45193.0   
WT                156599.333333   666943.333333         58592.0   
WT                147283.666667        615010.0         50895.0   
WT                     207490.0        761780.0         62390.5   
WT                166141.666667        745690.0         62122.0   
WT                211233.333333        866370.0         76912.0   
Mu                127722.666667   633516.666667         43149.5   
Mu                158869.666667   784453.333333         60037.5   
Mu                     156658.0   731076.666667         56974.5   
Mu                146465.666667   648163.333333         53222.5   
Mu                165979.333333        676740.0         53559.0   
Mu                191756.666667        787640.0         64556.0   
Mu                198803.333333   767766.666667         60326.5   
Mu                324696.666667       1081200.0         77807.0   

Main_ensembl_id ENSG00000144560  ...  ENSG00000167004 ENSG00000067955  \
Cell_type                        ...                                    
WT                      20133.0  ...        8698420.0   385000.333333   
WT                      24945.0  ...   8398763.333333   420642.666667   
WT                      21150.0  ...        6901480.0   389426.666667   
WT            

In [27]:
raw_Mean_tmpdf = vdata.TemporalDataFrame(
    raw_Mean_df,
    time_list=[
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
    ],
)

## 8 - replicates layers

In [28]:
raw_R1_df = proteomic_raw_df[
    proteomic_raw_df.columns[(proteomic_raw_df.columns.str[3:5] == "R1") == True]
]
raw_R1_df = raw_R1_df.transpose()
raw_R1_df.index = [cond[6:8] for cond in list(raw_R1_df.index)]
raw_R1_df.index = raw_R1_df.index.rename("Cell_type")
raw_R1_df = raw_R1_df.sort_values("Cell_type", ascending=False)
raw_R1_df

Main_ensembl_id  ENSG00000126653  ENSG00000124570  ENSG00000115677  \
Cell_type                                                            
WT                      112150.0          40426.0        1351100.0   
WT                      102800.0          44000.0        1524000.0   
WT                       93172.0          33604.0        1331200.0   
WT                      105610.0          42710.0        1618700.0   
WT                       94802.0          43591.0        1442700.0   
WT                      135300.0          40201.0        1634700.0   
WT                      116930.0          46099.0        1562400.0   
WT                      136980.0          52507.0        1934600.0   
Mu                      100250.0          34249.0        1226800.0   
Mu                      115760.0          43406.0        1585600.0   
Mu                      115860.0          38418.0        1535400.0   
Mu                      106860.0          39058.0        1366300.0   
Mu                      114260.0          37357.0        1337400.0   
Mu                      128690.0          40457.0        1485600.0   
Mu                      135740.0          40576.0        1370300.0   
Mu                      238950.0          48452.0        1885800.0   

Main_ensembl_id  ENSG00000110047  ENSG00000106588  ENSG00000011007  \
Cell_type                                                            
WT                      554030.0         632690.0          61753.0   
WT                      627240.0         690080.0          64703.0   
WT                      628120.0         587610.0          60229.0   
WT                      795010.0         662280.0          65747.0   
WT                      783310.0         592640.0          60929.0   
WT                      915460.0         624130.0          77827.0   
WT                      899380.0         622400.0          74710.0   
WT                     1154100.0         775110.0          80879.0   
Mu                      553380.0         614100.0          54773.0   
Mu                      689730.0         727490.0          70448.0   
Mu                      708150.0         731020.0          66392.0   
Mu                      671280.0         687820.0          64016.0   
Mu                      654750.0         670000.0          69876.0   
Mu                      727070.0         779360.0          77856.0   
Mu                      761180.0         737170.0          72864.0   
Mu                     1049200.0         825810.0         105900.0   

Main_ensembl_id  ENSG00000184047  ENSG00000204463  ENSG00000101974  \
Cell_type                                                            
WT                      201520.0         628140.0              NaN   
WT                      227240.0         676760.0              NaN   
WT                      185190.0         559950.0              NaN   
WT                      252360.0         659730.0              NaN   
WT                      236160.0         605890.0              NaN   
WT                      330750.0         747630.0              NaN   
WT                      268780.0         738600.0              NaN   
WT                      344300.0         849250.0              NaN   
Mu                      198140.0         608370.0              NaN   
Mu                      242760.0         742460.0              NaN   
Mu                      242950.0         706940.0              NaN   
Mu                      228150.0         629410.0              NaN   
Mu                      270590.0         657290.0              NaN   
Mu                      295360.0         765390.0              NaN   
Mu                      314270.0         769360.0              NaN   
Mu                      526800.0        1090700.0              NaN   

Main_ensembl_id  ENSG00000144560  ...  ENSG00000167004  ENSG00000067955  \
Cell_type                         ...                                     
WT                           NaN  ...        8534090.0         3884

In [29]:
raw_R2_df = proteomic_raw_df[
    proteomic_raw_df.columns[(proteomic_raw_df.columns.str[3:5] == "R2") == True]
]
raw_R2_df = raw_R2_df.transpose()
raw_R2_df.index = [cond[6:8] for cond in list(raw_R2_df.index)]
raw_R2_df.index = raw_R2_df.index.rename("Cell_type")

raw_R2_df = raw_R2_df.sort_values("Cell_type", ascending=False)
raw_R2_df

Main_ensembl_id  ENSG00000126653  ENSG00000124570  ENSG00000115677  \
Cell_type                                                            
WT                      124970.0              NaN        1008700.0   
WT                      132170.0              NaN        1162800.0   
WT                      116930.0              NaN        1006200.0   
WT                      136530.0              NaN        1217300.0   
WT                      125190.0              NaN        1100600.0   
WT                      174830.0              NaN        1235100.0   
WT                      142060.0              NaN        1169200.0   
WT                      160480.0              NaN        1452200.0   
Mu                      105900.0              NaN         933070.0   
Mu                      128400.0              NaN        1178600.0   
Mu                      121640.0              NaN        1147100.0   
Mu                      118980.0              NaN        1031400.0   
Mu                      134610.0              NaN         998670.0   
Mu                      156180.0              NaN        1124200.0   
Mu                      152030.0              NaN        1035400.0   
Mu                      261690.0              NaN        1385600.0   

Main_ensembl_id  ENSG00000110047  ENSG00000106588  ENSG00000011007  \
Cell_type                                                            
WT                      495220.0         480130.0          52919.0   
WT                      559490.0         523990.0          50537.0   
WT                      560330.0         468470.0          45839.0   
WT                      689750.0         509980.0          52662.0   
WT                      688950.0         454480.0          46164.0   
WT                      768090.0         470720.0          66253.0   
WT                      781550.0         458320.0          54614.0   
WT                      988840.0         596610.0          62045.0   
Mu                      455410.0         472690.0          49121.0   
Mu                      607300.0         574560.0          51026.0   
Mu                      631430.0         572120.0          55474.0   
Mu                      569230.0         527210.0          47689.0   
Mu                      582230.0         501500.0          56396.0   
Mu                      654830.0         596080.0          63704.0   
Mu                      674170.0         560730.0          62452.0   
Mu                      883520.0         625720.0          94033.0   

Main_ensembl_id  ENSG00000184047  ENSG00000204463  ENSG00000101974  \
Cell_type                                                            
WT                      127020.0         675210.0          30752.0   
WT                      147190.0         710850.0          37739.0   
WT                      115000.0         582130.0          31891.0   
WT                      136430.0         677710.0          41381.0   
WT                      128740.0         614070.0          35377.0   
WT                      161640.0         759610.0          42330.0   
WT                      138200.0         755470.0          42876.0   
WT                      171530.0         870300.0          53034.0   
Mu                      124920.0         647350.0          30307.0   
Mu                      161360.0         807930.0          40347.0   
Mu                      147110.0         742160.0          38856.0   
Mu                      140320.0         650180.0          36082.0   
Mu                      143980.0         678490.0          36262.0   
Mu                      179890.0         794570.0          45218.0   
Mu                      180330.0         780440.0          42040.0   
Mu                      256160.0        1071600.0          48114.0   

Main_ensembl_id  ENSG00000144560  ...  ENSG00000167004  ENSG00000067955  \
Cell_type                         ...                                     
WT                       19452.0  ...        8472370.0         4150

In [30]:
raw_R3_df = proteomic_raw_df[
    proteomic_raw_df.columns[(proteomic_raw_df.columns.str[3:5] == "R3") == True]
]
raw_R3_df = raw_R3_df.transpose()
raw_R3_df.index = [cond[6:8] for cond in list(raw_R3_df.index)]
raw_R3_df.index = raw_R3_df.index.rename("Cell_type")

raw_R3_df = raw_R3_df.sort_values("Cell_type", ascending=False)
raw_R3_df

Main_ensembl_id  ENSG00000126653  ENSG00000124570  ENSG00000115677  \
Cell_type                                                            
WT                      108550.0              NaN        1667200.0   
WT                      111670.0              NaN        1878900.0   
WT                       85498.0              NaN        1632500.0   
WT                       99239.0              NaN        1984500.0   
WT                       92848.0              NaN        1764200.0   
WT                      136290.0              NaN        1988400.0   
WT                      118650.0              NaN        1854200.0   
WT                      133270.0              NaN        2400300.0   
Mu                       92343.0              NaN        1490300.0   
Mu                      119310.0              NaN        1939800.0   
Mu                      110050.0              NaN        1851500.0   
Mu                       96349.0              NaN        1697200.0   
Mu                      112510.0              NaN        1653100.0   
Mu                      128080.0              NaN        1844300.0   
Mu                      134610.0              NaN        1677400.0   
Mu                      214620.0              NaN        2280200.0   

Main_ensembl_id  ENSG00000110047  ENSG00000106588  ENSG00000011007  \
Cell_type                                                            
WT                      509650.0         474270.0          55748.0   
WT                      549440.0         544550.0          55908.0   
WT                      569680.0         460280.0          49921.0   
WT                      692700.0         521490.0          60577.0   
WT                      679770.0         459200.0          52500.0   
WT                      776850.0         484740.0          75984.0   
WT                      752200.0         485480.0          62077.0   
WT                      978810.0         617400.0          71590.0   
Mu                      460290.0         478480.0          55617.0   
Mu                      590780.0         560900.0          53719.0   
Mu                      637070.0         574990.0          59129.0   
Mu                      590670.0         540440.0          58232.0   
Mu                      588170.0         519630.0          63970.0   
Mu                      637960.0         598120.0          74224.0   
Mu                      645690.0         569490.0          68116.0   
Mu                      848520.0         638210.0         109470.0   

Main_ensembl_id  ENSG00000184047  ENSG00000204463  ENSG00000101974  \
Cell_type                                                            
WT                       56867.0         658250.0          57563.0   
WT                       66622.0         726070.0          73614.0   
WT                       57868.0         576660.0          58495.0   
WT                       81008.0         663390.0          75803.0   
WT                       76951.0         625070.0          66413.0   
WT                      130080.0         778100.0          82451.0   
WT                       91445.0         743000.0          81368.0   
WT                      117870.0         879560.0         100790.0   
Mu                       60108.0         644830.0          55992.0   
Mu                       72489.0         802970.0          79728.0   
Mu                       79914.0         744130.0          75093.0   
Mu                       70927.0         664900.0          70363.0   
Mu                       83368.0         694440.0          70856.0   
Mu                      100020.0         802960.0          83894.0   
Mu                      101810.0         753500.0          78613.0   
Mu                      191130.0        1081300.0         107500.0   

Main_ensembl_id  ENSG00000144560  ...  ENSG00000167004  ENSG00000067955  \
Cell_type                         ...                                     
WT                       20814.0  ...        9088800.0         3515

In [31]:
raw_R1_tmpdf = vdata.TemporalDataFrame(
    raw_R1_df,
    time_list=[
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
    ],
)
raw_R2_tmpdf = vdata.TemporalDataFrame(
    raw_R1_df,
    time_list=[
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
    ],
)
raw_R3_tmpdf = vdata.TemporalDataFrame(
    raw_R1_df,
    time_list=[
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
        "5D",
        "6D",
        "7D",
        "8D",
        "9D",
        "10D",
        "12D",
        "14D",
    ],
)

## Vdata

In [32]:
raw_proteomic = {
    "R1": raw_R1_tmpdf,
    "R2": raw_R2_tmpdf,
    "R3": raw_R3_tmpdf,
    "Mean": raw_Mean_tmpdf,
}

In [33]:
obs = TemporalDataFrame(
    {"Cell_type": raw_R1_tmpdf.index},
    time_list=[
        "5D",
        "5D",
        "6D",
        "6D",
        "7D",
        "7D",
        "8D",
        "8D",
        "9D",
        "9D",
        "10D",
        "10D",
        "12D",
        "12D",
        "14D",
        "14D",
    ],
    index=[
        "WT",
        "Mu",
        "WT",
        "Mu",
        "WT",
        "Mu",
        "WT",
        "Mu",
        "WT",
        "Mu",
        "WT",
        "Mu",
        "WT",
        "Mu",
        "WT",
        "Mu",
    ],
    name="obs",
)

In [34]:
proteomic_raw_vdata = VData(raw_proteomic, obs=obs, var=proteomic_var_df)

In [35]:
proteomic_raw_vdata.var

id    short_name      organism
Main_ensembl_id                                       
ENSG00000126653          0  NSRP1;CCDC55  Homo sapiens
ENSG00000124570          1      SERPINB6  Homo sapiens
ENSG00000115677          2         HDLBP  Homo sapiens
ENSG00000110047          3          EHD1  Homo sapiens
ENSG00000106588          4         PSMA2  Homo sapiens
...                    ...           ...           ...
ENSG00000198467  2630;4028          TPM2  Homo sapiens
ENSG00000120802  3074;3075          TMPO  Homo sapiens
ENSG00000159352  3281;4053         PSMD4  Homo sapiens
ENSG00000165119  3362;4016        HNRNPK  Homo sapiens
ENSG00000131236  3494;4000          CAP1  Homo sapiens

[5360 rows x 3 columns]

In [36]:
proteomic_raw_vdata.obs["14D"]

View of TemporalDataFrame 'obs'
Time point : 14.0 days
   Cell_type
WT        WT
Mu        Mu

[2 x 1]


In [37]:
type(proteomic_raw_vdata.layers["R1"]["5D", "WT", "ENSG00000101974"].values[0, 0])

numpy.float64

In [44]:
import numpy as np
proteomic_raw_vdata.layers["Mean"].astype(np.float64)

In [45]:
setLoggingLevel('INFO')
proteomic_raw_vdata.write("./../results/Proteomic_VData.h5")

16:25:51 :       INFO : [write.py] Saving dict layers
16:25:51 :       INFO : [write.py]   ↳ Saving TemporalDataFrame R1
16:25:51 :       INFO : [write.py]     ↳ Saving Series index
16:25:51 :       INFO : [write.py]     ↳ Saving None value for time_col_name
16:25:51 :       INFO : [write.py]     ↳ Saving Series columns
16:25:51 :       INFO : [write.py]     ↳ Saving time point 5.0D
16:25:52 :       INFO : [write.py]     ↳ Saving time point 6.0D
16:25:52 :       INFO : [write.py]     ↳ Saving time point 7.0D
16:25:52 :       INFO : [write.py]     ↳ Saving time point 8.0D
16:25:53 :       INFO : [write.py]     ↳ Saving time point 9.0D
16:25:53 :       INFO : [write.py]     ↳ Saving time point 10.0D
16:25:54 :       INFO : [write.py]     ↳ Saving time point 12.0D
16:25:54 :       INFO : [write.py]     ↳ Saving time point 14.0D
16:25:54 :       INFO : [write.py]   ↳ Saving TemporalDataFrame R2
16:25:54 :       INFO : [write.py]     ↳ Saving Series index
16:25:54 :       INFO : [write.py] 

## 9 - test VData reading

In [ ]:
vdata_prot_read = vdata.read_from_csv("./../results/Proteomic_VData")

In [ ]:
vdata_prot_read.var

In [ ]:
vdata_prot_read.layers["Mean"]